Increase accumulate_grad_ gradient's expected refcount to account for pybind

xmfan opened the pull request:
Account for pybind of the op holding 1 ref when torch.ops.inductor.accumulate_grad_.default is called during run time

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119068
* #119334
* #118817



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

xmfan committed:
Increase gradient's expected refcount from 1 to 2 for accumulate_grad_ inductor op

[ghstack-poisoned]

jansel reviewed (changes requested):
Seems fine, but tests are failing.

xmfan committed:
Update on "Increase gradient's expected refcount from 1 to 2 for accumulate_grad_ inductor op"

Account for pybind of the op holding 1 ref when torch.ops.inductor.accumulate_grad_.default is called during run time




[ghstack-poisoned]

xmfan committed:
Update on "Increase gradient's expected refcount from 1 to 2 for accumulate_grad_ inductor op"

Account for pybind of the op holding 1 ref when torch.ops.inductor.accumulate_grad_.default is called during run time




[ghstack-poisoned]

xmfan committed:
Update on "Increase gradient's expected refcount from 1 to 2 for accumulate_grad_ inductor op"

Account for pybind of the op holding 1 ref when torch.ops.inductor.accumulate_grad_.default is called during run time




[ghstack-poisoned]

xmfan committed:
Update on "Increase gradient's expected refcount from 1 to 2 for accumulate_grad_ inductor op"

Account for pybind of the op holding 1 ref when torch.ops.inductor.accumulate_grad_.default is called during run time




[ghstack-poisoned]

xmfan committed:
Update on "Increase accumulate_grad_ gradient's expected refcount to account for pybind"

Account for pybind of the op holding 1 ref when torch.ops.inductor.accumulate_grad_.default is called during run time




[ghstack-poisoned]

xmfan committed:
Update on "Increase accumulate_grad_ gradient's expected refcount to account for pybind"

Account for pybind of the op holding 1 ref when torch.ops.inductor.accumulate_grad_.default is called during run time




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

xmfan commented:
@pytorchbot merge
